### PR TITLE
Add manually triggered action for pushing docker image by SHA

### LINKED ISSUE #597: Publish a docker image on each (non draft) PR commit
## Problem

We are currently only publishing tagged docker images to GHCR when a release is cut. When we are deploying, (currently only with the ingestion server/indexer workers, but ideally with the API down the road) we have the option to deploy a specific docker image tag. We can point the API to a specific commit and deploy said commit (in staging, for instance). We do not currently have an allegory for the docker images.

## Description

We should add logic to the CI/CD to push the docker image tagged with the commit **on every PR commit**. That way individual commits can be used for deployments rather than having to cut a release in order to test something in staging.

While this might seem like it would pollute GHCR with tons of images, the reality is that a new image would be pushed only if changes to the docker image have occurred! In most of our code-change cases, the produced docker image has the exact same hash as the one published before it. Example: https://github.com/wordpress/openverse-api/pkgs/container/openverse-api/17792291?tag=2b3d93521811ffd4554f3ad44db6b585401f41e2 _(Note the `Other tags on this version` section in the bottom right)_.

We should probably restrict this step to non-draft PRs; there's no sense in pushing a docker image on a PR that we know is still a work in progress.

## Additional context

For one possible implementation, see: 6abdcbdd523b154f37b393fb9a24e2e3cf9d0870

We may also want to do something similar for the catalog.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #597 by @AetherUnbound

## Description

This PR adds the ability to tag a Docker image artifact by SHA only and push it to our container registry. The workflow is manually triggered and requires an input on which image to push (only one option is allowed because we are almost never making changes to _both_ the API and ingestion server images). 

The example use case for this: We've made a change to one of the services, and want to deploy it on staging. Our deployment process requires (or will soon require) a tagged Docker image to deploy. Running this workflow will tag and push an image by SHA only, so `latest` is unaffected. We can then deploy by pointing to said commit.

## Testing Instructions

I'm going to test this in this PR!

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by AetherUnbound** [comment]
This is ready for review! I ran an API job for commit: `4d4a466585d9a55441f708fab540ecb92ae5523f` (https://github.com/WordPress/openverse-api/pull/636/commits/4d4a466585d9a55441f708fab540ecb92ae5523f) using `gh workflow run push_docker_image.yml --ref feature/docker-image-when-needed#597 -f image=api`.

Here is the run: https://github.com/WordPress/openverse-api/runs/5891296828?check_suite_focus=true
Here is the GHCR image tagged with the commit: https://github.com/wordpress/openverse-api/pkgs/container/openverse-api/18665267?tag=4d4a466585d9a55441f708fab540ecb92ae5523f (note that `latest` is _not_ on this build).

File: .github/workflows/push_docker_image.yml
Comment (sarayourfriend) [inline .github/workflows/push_docker_image.yml:31]: I guess the only thing to note here is Github's artifact retention policy, [which defaults to 90 days](https://docs.github.com/en/organizations/managing-organization-settings/configuring-the-retention-period-for-github-actions-artifacts-and-logs-in-your-organization). So this action could fail if run on a SHA that hasn't been built for the last 90 days.

That seems unlikely, but just in case, something to note!

**Review by sarayourfriend — APPROVED** [review]
LGTM!

**Review by obulat — APPROVED** [review]
Thank you for detailed explanation in this PR! It would be nice to have it somewhere in the documentation, if it's not already (along with the comment about 90 days that Sara mentioned)

File: .github/workflows/push_docker_image.yml
Comment (AetherUnbound) [reply .github/workflows/push_docker_image.yml:31]: That's a great point! I hope we're not trying to run this on a 90-day old SHA 😅

**Comment by AetherUnbound** [comment]
I'll add some documentation for this, then merge it 🙂

**Comment by AetherUnbound** [comment]
I added documentation and made some minor adjustments to other pages. I'd like to add screenshots but until this gets merged into `main`, the actions described won't be available. We can do that in a follow up PR. I'll merge this once everything's done building!

**Pull request merged**
